code/script.js

```diff
@@ -218,108 +248,246 @@ const generateBoard = () => {
+  let formattedMS = ms.toString().padStart(2, "0");
+
+  return `Your playtime is: ${formattedHH}:${formattedMM}:${formattedSS}`; // Returns a string
+}
+
+// A function that modifies the innerHTML
+const print = (txt) => {
+  document.getElementById("timePlayed").innerHTML = txt;
 }
 
+// A function that starts the timer
+const startTimer = () => {
+  startTime = Date.now() - elapsedTime;
+  timerInterval = setInterval(function printTime() {
+    elapsedTime = Date.now() - startTime;
+    print(timeToString(elapsedTime));
+  }, 10);
+};
+
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const category = questions.options[questions.selectedIndex].parentNode.label;
+  const value = questions.value;
 
-  currentQuestion = {
+  currentQuestion = { // creating the object that holds information about what I pick from the dropdown
     category: category,
-    // value: value
-  }
-}
+    value: value,
+  };
+};
+
 
-// This function should be invoked when you click on 'Find Out' button.
+// A function that checks which I picked in the dropdown 
+// It then takes the value and compares it to the secret persons value
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+  // This if statements first checks which category is picked
+  if (category === "eyes" || category === "gender") {
+    // If the category is hair/eyes it gets the value in this way
+    if (secret[category] === value) {
+      // If the value is true it runs filterCharacters with a true value
+      filterCharacters(true);
+    } else {
+      filterCharacters(); // Otherwise it runs with a false value
+    }
+  } else if (category === "accessories" || category === "other" || category === "hair") {
+    // If the category is accessories/other it gets the value with the includes() method since their values are stored in an array
+    if (secret[category].includes(value)) {
+      // If the value is true it runs filterCharacters with a true value
+      filterCharacters(true);
+    } else {
+      filterCharacters();
+    }
+  }
 
-  } else if (category === 'accessories' || category === 'other') {
+  document.getElementById("counter").innerHTML = `This far you have made ${counter} guesses`; // prints this when the user makes a guess
+  counter++; // then adds 1 to the counter
```

Comment:
I like this simple approach you took with the counter. Very basic and it gets teh job done perfectly.